api/include/opentelemetry/logs/logger.h

```diff
@@ -55,65 +54,322 @@ class Logger
+  {
+    Log(Severity::kTrace, attributes);
+  }
+
+  /**
+   * Writes a log with a severity of trace.
+   * @param name The name of the log
+   * @param attributes A key/value object
+   */
+  template <class T,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
+  inline void Trace(nostd::string_view name, const T &attributes) noexcept
+  {
+    Log(Severity::kTrace, name, attributes);
+  }
+
+  /**
+   * Writes a log with a severity of debug.
+   * @param message The message to log
+   */
+  inline void Debug(nostd::string_view message) noexcept { Log(Severity::kDebug, message); }
+
+  /**
+   * Writes a log with a severity of debug.
+   * @param name The name of the log
+   * @param message The message to log
+   */
+  inline void Debug(nostd::string_view name, nostd::string_view message) noexcept
+  {
+    Log(Severity::kDebug, name, message);
+  }
+
+  /**
+   * Writes a log with a severity of debug.
+   * @param attributes A key/value object
+   */
+  template <class T,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
+  inline void Debug(const T &attributes) noexcept
+  {
+    Log(Severity::kDebug, attributes);
+  }
+
+  /**
+   * Writes a log with a severity of debug.
+   * @param name The name of the log
+   * @param attributes A key/value object
+   */
+  template <class T,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
+  inline void Debug(nostd::string_view name, const T &attributes) noexcept
+  {
+    Log(Severity::kDebug, name, attributes);
+  }
+
+  /**
+   * Writes a log with a severity of info.
+   * @param message The message to log
+   */
+  inline void Info(nostd::string_view message) noexcept { Log(Severity::kInfo, message); }
+
+  /**
+   * Writes a log with a severity of info.
+   * @param name The name of the log
+   * @param message The message to log
+   */
+  inline void Info(nostd::string_view name, nostd::string_view message) noexcept
+  {
+    Log(Severity::kInfo, name, message);
+  }
+
+  /**
+   * Writes a log with a severity of info.
+   * @param attributes A key/value object
+   */
+  template <class T,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
+  inline void Info(const T &attributes) noexcept
+  {
+    Log(Severity::kInfo, attributes);
+  }
+
+  /**
+   * Writes a log with a severity of info.
+   * @param name The name of the log
+   * @param attributes A key/value object
+   */
+  template <class T,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
+  inline void Info(nostd::string_view name, const T &attributes) noexcept
+  {
+    Log(Severity::kInfo, name, attributes);
+  }
+
+  /**
+   * Writes a log with a severity of warn.
+   * @param message The message to log
+   */
+  inline void Warn(nostd::string_view message) noexcept { Log(Severity::kWarn, message); }
```

Comment:
Seems name `Warning` is more practical than `Warn`?

Reply:
`Warn` seems to be used by the [Log Data Model](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/logs/data-model.md#field-severitynumber), and looks like [Log4cxx](http://www.yolinux.com/TUTORIALS/Log4cxx.html) and [Log4J](https://www.tutorialspoint.com/log4j/log4j_logging_levels.htm) also do it this way